torch/csrc/shim_common.cpp

```diff
@@ -110,12 +115,24 @@ static c10::IValue to_ivalue(
           stable_ivalue, extension_build_version));
     }
     case c10::TypeKind::DeviceObjType: {
-      auto deleter = [](DeviceOpaque* device) { torch_delete_device(device); };
-      auto device_ptr = std::unique_ptr<DeviceOpaque, decltype(deleter)>(
-          torch::stable::detail::_to<DeviceHandle>(
-              stable_ivalue, extension_build_version),
-          deleter);
-      return c10::IValue(*reinterpret_cast<c10::Device*>(device_ptr.get()));
+      // Unpack device type and index from StableIValue
+      // Lower 8 bits = device type, upper 32 bits = device index
+      c10::DeviceType device_type = static_cast<c10::DeviceType>(
+          static_cast<int8_t>(stable_ivalue & 0xFF));
+      int32_t device_index =
+          static_cast<int32_t>((stable_ivalue >> 32) & 0xFFFFFFFF);
+      TORCH_CHECK(
```

Comment:
Why do we want this check? Does this mean that we need to remove this check once we support more than 128 GPUs?